src/app/body/all-products/all-products.component.ts

```diff
@@ -1,31 +1,44 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Product } from '../../models/product.model';
 import { ProductService } from '../../services/product.service';
-import { ActivatedRoute } from '@angular/router';
+import { WishlistService } from '../../services/wishlist.service';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-all-products',
   templateUrl: './all-products.component.html',
   styleUrls: ['./all-products.component.scss']
 })
-export class AllProductsComponent implements OnInit {
+export class AllProductsComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject();
   public products: Product[];
   public page: number;
   public productsToShow = 5;
+  public productsCount: number;
 
   constructor(private productService: ProductService,
-              private route: ActivatedRoute) { }
+              private wishlistService: WishlistService) { }
 
-  ngOnInit(): void {
-    this.productService.products.subscribe(products => {
-      this.products = products;
-    });
+  async ngOnInit(): Promise<void> {
+    await this.wishlistService.getWishlist().then();
```

Comment:
This logic is not used in this component, so why do you need it here?

Reply:
On my opinion you have too complicated `async ngOnInit(): Promise<void>` why this onInit returns Promise and so on... 
This `ngOnInit` should be as simple as possible and not to make side effects and to do some logic, that not related to component, that called from.

Reply:
These logic get the wishlist from firebase, which I need to have on these pages: all-products, product-detail, cart and wishlist
and without async it does not work properly, I found such a solution on forums
it returns Promise because linter requires it when async is used
But I know some way to avoid it, will try it later